packages/react-components/src/PasswordStrength.tsx

```diff
@@ -1,52 +1,78 @@
 // Copyright 2017-2020 @polkadot/react-components authors & contributors
 // SPDX-License-Identifier: Apache-2.0
 
+import { useTranslation } from '@polkadot/react-components/translate';
 import React from 'react';
 import styled from 'styled-components';
+import strengthTester from 'owasp-password-strength-test';
+
+const MAX_STRENGTH = 7; // equal to number of password tests in owasp strength tester
 
 interface Props {
   className?: string;
   value: string;
 }
 
 function calcStrength (password: string): number {
-  // at least 8 characters
-  return (/.{8,}/.test(password) ? 1 : 0) * (
-    // bonus if longer
-    (/.{12,}/.test(password) ? 1 : 0) +
-    // a lower letter
-    (/[a-z]/.test(password) ? 1 : 0) +
-    // a upper letter
-    (/[A-Z]/.test(password) ? 1 : 0) +
-    // a digit
-    (/\d/.test(password) ? 1 : 0) +
-    // a special character
-    (/[^A-Za-z0-9]/.test(password) ? 1 : 0)
-  );
+  strengthTester.config({
+    allowPassphrases: true,
+    maxLength: 128,
+    minLength: 8,
+    minPhraseLength: 20
+  });
+  const testResult = strengthTester.test(password);
+  const passedTests = Math.max(0, testResult.passedTests.length - testResult.failedTests.length);
+
+  return testResult.isPassphrase ? MAX_STRENGTH : passedTests;
 }
 
 function PasswordStrength ({ className = '', value }: Props): React.ReactElement<Props> {
-  // No need for memo, component is already memo-ed (only changes on value)
-  const style = { width: `${calcStrength(value) * 20}%` };
+  const { t } = useTranslation();
+
+  const passwordStrength = calcStrength(value);
+  const style = { width: `${passwordStrength * 100 / MAX_STRENGTH}%` };
 
   return (
-    <div className={className}>
-      <div
-        className='highlight--bg'
-        style={style}
-      />
+    <div
+      className={className}
+      style={{ display: (value ? 'flex' : 'none') }}
+    >
+      {t<string>('weak')}
+      <div className='ui--Strength-bar'>
+        <div
+          className='ui--Strength-bar-highlighted'
+          style={style}
+        />
+      </div>
+      {t<string>('strong')}
     </div>
   );
 }
 
 export default React.memo(styled(PasswordStrength)`
-  background: ##eee6e6;
-  border-radius: 0.25rem;
-  margin-left: 2rem;
-  margin-top: 0.25rem;
-  overflow: hidden;
-
-  > div {
-    height: 0.5rem;
-  }
+align-items: center;
+margin-top: 0.5rem;
+margin-left: 2rem;
+font-weight: 800;
+font-size: 10px;
+line-height: 14px;
```

Comment:
em please, so we have consistency in sizes. (We don't have this size anywhere else)